### PR TITLE
Adding egress IPv6 for node-local-dns queries

### PR DESCRIPTION
**Pull request by raviranjanelastisys**
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind feature


**What this PR does / why we need it**:
We found some error in the `node-local-dns` pod for IPv6 queries - 
```
[ERROR] plugin/errors: 2 alertmanager-kube-prometheus-stack-alertmanager-0.alertmanager-operated. ANY: dial tcp [2620:fe::fe]:53: i/o timeout
[ERROR] plugin/errors: 2 5185845442857603503.5577735529860551140.cluster.local. HINFO: dial tcp 10.233.3.23:53: i/o timeout
[ERROR] plugin/errors: 2 6681947942670949562.3324415659505387376.in-addr.arpa. HINFO: dial tcp 10.233.3.23:53: i/o timeout
[ERROR] plugin/errors: 2 5216755221347913579.934064619013292502.ip6.arpa. HINFO: dial tcp 10.233.3.23:53: i/o timeout
```
and allowing the Ipv6 egress fixed the problem in firewall

**Tested**:

```
openstack security group show kubespray-2-22-wc-k8s

| revision_number | 10                                                                                                                                                                                                                                                                                                                         |
| rules           | created_at='2023-08-27T14:18:10Z', direction='egress', ethertype='IPv6', id='0a25dd23-3b94-468f-9dd0-b906249566f3', normalized_cidr='::/0', protocol='udp', remote_ip_prefix='::/0', standard_attr_id='18272950', updated_at='2023-08-27T14:18:10Z'                                                                        |
|                 | created_at='2023-08-27T14:18:10Z', direction='egress', ethertype='IPv6', id='571a7633-e808-40b0-adce-153e0a633383', normalized_cidr='::/0', protocol='tcp', remote_ip_prefix='::/0', standard_attr_id='18272947', updated_at='2023-08-27T14:18:10Z'    
```

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
No
```

### CODE REVIEW

**Review by Xartos — COMMENTED** [review]
I feel like if we're going to add support for IPv6 we should add it for all the rules.
~But is this something that we actually need? Or is it just for future proofing?~

Sorry, didn't read the PR message properly... But can't we just disable IPv6 for node-local-dns instead? We don't have any support for it anywhere else

**Comment by raviranjanelastisys** [comment]
> I feel like if we're going to add support for IPv6 we should add it for all the rules. ~But is this something that we actually need? Or is it just for future proofing?~
> 
> Sorry, didn't read the PR message properly... But can't we just disable IPv6 for node-local-dns instead? We don't have any support for it anywhere else

Yeah i can check this options, Meanwhile i will open the mini-ADR to discuss too :)

File: contrib/terraform/openstack/variables.tf [hunk 1]
```diff
@@ -220,30 +220,60 @@ variable "bastion_allowed_remote_ips" {
...
+}
+
 variable "k8s_allowed_remote_ips" {
   description = "An array of CIDRs allowed to SSH to hosts"
   type        = list(string)
   default     = []
 }
 
+variable "k8s_allowed_remote_ips_ipv6" {
+  description = "An array of CIDRs allowed to SSH to hosts"
```
Comment (Xartos) [inline contrib/terraform/openstack/variables.tf:248]:
```suggestion
  description = "An array of IPv6 CIDRs allowed to SSH to hosts"
```

File: contrib/terraform/openstack/variables.tf [hunk 2]
```diff
@@ -220,30 +220,60 @@ variable "bastion_allowed_remote_ips" {
...
+}
+
 variable "master_allowed_remote_ips" {
   description = "An array of CIDRs allowed to access API of masters"
   type        = list(string)
   default     = ["0.0.0.0/0"]
 }
 
+variable "master_allowed_remote_ipv6_ips" {
+  description = "An array of CIDRs allowed to access API of masters"
```
Comment (Xartos) [inline contrib/terraform/openstack/variables.tf:236]:
```suggestion
  description = "An array of IPv6 CIDRs allowed to access API of masters"
```

File: contrib/terraform/openstack/variables.tf [hunk 3]
```diff
@@ -220,30 +220,60 @@ variable "bastion_allowed_remote_ips" {
   default     = ["0.0.0.0/0"]
 }
 
+variable "bastion_allowed_remote_ipv6_ips" {
+  description = "An array of CIDRs allowed to SSH to hosts"
```
Comment (Xartos) [inline contrib/terraform/openstack/variables.tf:224]:
```suggestion
  description = "An array of IPv6 CIDRs allowed to SSH to hosts"
```

File: contrib/terraform/openstack/variables.tf [hunk 4]
```diff
@@ -257,12 +287,32 @@ variable "worker_allowed_ports" {
   ]
 }
 
+variable "worker_allowed_ports_ipv6" {
+  type = list(any)
+
+  default = [
+    {
+      "ethertype"        = "IPv6"
```
Comment (Xartos) [inline contrib/terraform/openstack/variables.tf:295]: This shouldn't be here
```suggestion
```

File: contrib/terraform/openstack/modules/compute/main.tf [hunk 5]
```diff
@@ -131,6 +211,24 @@ resource "openstack_networking_secgroup_rule_v2" "egress" {
   security_group_id = openstack_networking_secgroup_v2.k8s.id
 }
 
+resource "openstack_networking_secgroup_rule_v2" "egress_ipv6_tcp" {
+  count             = length(var.k8s_allowed_egress_ipv6_ips)
+  direction         = "egress"
+  ethertype         = "IPv6"
+  protocol          = "tcp"
```
[the remark this answers is not in the document]
Comment (raviranjanelastisys) [reply contrib/terraform/openstack/modules/compute/main.tf]: Thanks, I removed it and tested and updated :)

File: contrib/terraform/openstack/variables.tf (hunk 1 above)
Comment (raviranjanelastisys) [reply contrib/terraform/openstack/variables.tf]: Updated

File: contrib/terraform/openstack/variables.tf (hunk 2 above)
Comment (raviranjanelastisys) [reply contrib/terraform/openstack/variables.tf]: Updated

File: contrib/terraform/openstack/variables.tf (hunk 3 above)
Comment (raviranjanelastisys) [reply contrib/terraform/openstack/variables.tf]: Updated

File: contrib/terraform/openstack/variables.tf (hunk 4 above)
Comment (raviranjanelastisys) [reply contrib/terraform/openstack/variables.tf]: Updated

**Review by Xartos — APPROVED** [review]

**Comment by raviranjanelastisys** [comment]
Closing this PR as we don't need here , and opened up in the correct repository - https://github.com/elastisys/compliantkubernetes-kubespray/pull/316

**Pull request closed without merging**